src/unix.rs

```diff
@@ -284,19 +288,15 @@ impl Client {
 
     pub fn try_acquire(&self) -> io::Result<Option<Acquired>> {
         let mut buf = [0];
+        let mut fifo = &self.read;
 
-        let (mut fifo, is_non_blocking) = match self {
-            Self::Fifo {
-                file,
-                is_non_blocking,
-                ..
-            } => (file, is_non_blocking),
-            _ => return Err(io::ErrorKind::Unsupported.into()),
-        };
-
-        if !is_non_blocking.load(Ordering::Relaxed) {
-            set_nonblocking(fifo.as_raw_fd(), true)?;
-            is_non_blocking.store(true, Ordering::Relaxed);
+        if let Some(is_non_blocking) = self.is_non_blocking.as_ref() {
+            if !is_non_blocking.load(Ordering::Relaxed) {
+                set_nonblocking(fifo.as_raw_fd(), true)?;
+                is_non_blocking.store(true, Ordering::Relaxed);
+            }
+        } else {
```

Comment:
the check isn't quite correct since an inherited FD from make may already be in nonblocking mode. We can't modify the flag when sharing the file description with other processes, but we can still opportunistically use it.

Which also means the `is_non_blocking` name is misleading.

Reply:
> but we can still opportunistically use it.

We can't do it, since it is shared between processes, other processes can clear the non-blocking flags and convert it to blocking at anytime.

It can happen after we checked that it has non-blocking set and start a non-blocking read, or it can happen during the read, which I'm not sure how the kernel will handle it but it's pretty bad.

Reply:
Other processes shouldn't be manipulating that flag on a shared resource. But yeah, I guess it's better to be on the safe side.